Library/Homebrew/test/command_options_spec.rb

```diff
@@ -0,0 +1,53 @@
+require "command_options"
+
+describe Homebrew::CommandOptions do
+  describe "Tests for CommandOptions() class: " do
+    it "checks correctness of initialize() method" do
+      command_options = Homebrew::CommandOptions.new("test command")
+      expect(command_options.command_name).to eq("test command")
+      expect(command_options.valid_options).to eq({})
+    end
+
+    it "checks correct setting of valid_options" do
+      command_options = Homebrew::CommandOptions.new("test command")
+      command_options.option("--bar")
+      command_options.option("--foo", "do foo")
+      command_options.option("--quiet", "be quiet")
+      expect(command_options.valid_options).to eq("--foo"=>"do foo", "--quiet"=>"be quiet",
+        "--bar"=>"No description for this option is available")
+    end
+
+    it "checks error message if invalid option(s) are provided" do
+      command_options = Homebrew::CommandOptions.new("test command")
+      command_options.option("--bar")
+      argv_options = ["--foo"]
+      error_message = command_options.get_error_message(argv_options)
+      expect(error_message).to include("<test command> has only 1 valid option: --bar")
+
+      command_options.option("--foo", "do foo")
+      command_options.option("--quiet", "be quiet")
+      argv_options = ["--bar1", "--bar2", "--bar1", "--bar", "--foo"]
+      error_message = command_options.get_error_message(argv_options)
+      expected_error_message = <<-EOS.undent
```

Comment:
I think this would be easier to read without the temporary variable — just embed the start of the heredoc into the method call.